regression-test/suites/datatype_p0/decimalv3/test_decimalv3_where.groovy

```diff
@@ -23,4 +23,16 @@ suite("test_decimalv3_where") {
 	sql """CREATE TABLE ${tableName} ( `id` varchar(11) NULL COMMENT '唯一标识', `name` varchar(10) NULL COMMENT '采集时间', `age` int(11) NULL, `dr` decimalv3(10, 0) ) ENGINE=OLAP UNIQUE KEY(`id`) COMMENT 'test' DISTRIBUTED BY HASH(`id`) BUCKETS 10 PROPERTIES ( "replication_allocation" = "tag.location.default: 1", "in_memory" = "false", "storage_format" = "V2", "light_schema_change" = "true", "disable_auto_compaction" = "false" );"""
 	sql """insert into ${tableName} values (1,'doris',20,324.10),(2,'spark',10,95.5),(3,'flink',9,20)"""
 	qt_decimalv3 "select * from ${tableName} where dr != 1  order by age;"
+
+    sql """set enable_nereids_planner=false;"""
+    sql """drop table if exists test_sys_update_basic_test_update_decimal_tb"""
+    sql """CREATE TABLE test_sys_update_basic_test_update_decimal_tb (
+              k1 DECIMAL(10, 5) NULL, 
+              v1 DECIMAL(10, 5) NULL
+            ) UNIQUE KEY(k1) DISTRIBUTED BY HASH(k1) BUCKETS 5 PROPERTIES (
+              "replication_allocation" = "tag.location.default: 1"
+            );"""
+    sql """insert into test_sys_update_basic_test_update_decimal_tb values
+                (1.001, 2.002), (1.002, 0.00000002), (1.003, 0.100000001), (1.004, 0.100044001), (1.005, 0.100045001);"""
+    qt_select """select * from test_sys_update_basic_test_update_decimal_tb where k1 = 1.001;"""
```

Comment:
after this pr, does we also support "where k1 = 1.001" for `UPDATE` stmt?

If we also support `UPDATE` stmt, could you add one case for `UPDATE` stmt?

The problem desc in pr https://github.com/apache/doris/pull/32246 :
```
    // need to use "1.001"(quoted) because non-nereids dml seems treat 1.001 as double,
    // which will cause wrong result of decimal comparision
    sql """
        UPDATE test_sys_update_basic_test_update_decimal_tb SET v1="0.00000001" WHERE k1 = "1.001";
    """
```

Reply:
case is added